src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -728,6 +784,8 @@ VarAndSeq randomVariable(SimpleList<Sequence> candidates, Type inputType, boolea
       Sequence chosenSeq;
       if (GenInputsAbstract.small_tests) {
         chosenSeq = Randomness.randomMemberWeighted(candidates);
+      } else if (GenInputsAbstract.enable_orienteering) {
```

Comment:
The lack of parallelism between the above case and this one bothers me.
I think it would be better to make both use an external map, or to make both store a sequence's weight in the sequence (`getWeight` would just look this up, and the current implementation would be abstracted out just as `orienteeringWeight` is abstracted out.  But the code that uses one or the other would be otherwise the same.

How does this relate to other changes that are currently being pull-requested?  Does it overlap with them, or is it orthogonal?